src/gui/layertree/qgslayertreeview.cpp

```diff
@@ -95,10 +95,11 @@ void QgsLayerTreeView::setModel( QAbstractItemModel *model )
          );
 
   mProxyModel = new QgsLayerTreeProxyModel( treeModel, this );
-
-  connect( mProxyModel, &QAbstractItemModel::rowsInserted, this, &QgsLayerTreeView::modelRowsInserted );
-  connect( mProxyModel, &QAbstractItemModel::rowsRemoved, this, &QgsLayerTreeView::modelRowsRemoved );
-
+  if ( connectProxy )
+  {
+    connect( mProxyModel, &QAbstractItemModel::rowsInserted, this, &QgsLayerTreeView::modelRowsInserted );
```

Comment:
@nyalldawson This was the cause of the crashes, upon adding a legend node in the main view this would also be called in the custom view, in the instances where the model didn't have any 'spare' rows ( no nodes removed previously or other state), this would fetch data from an invalid node (as the good one would be added already). This is why it lead to crashes. I made the parameter optional to maintain everything functionnal.